fintrack/src/components/TransactionCreate.js

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useContext } from 'react'
+import { useNavigate, useLocation } from 'react-router-dom'
+
+import { createTransaction } from '../utils/api-transactions'
+
+import UserContext from '../context/UserContext'
+import { AuthContext } from '../context/AuthContext'
+
+function TransactionCreationPage() {
+
+    const navigate = useNavigate()
```

Comment:
There is a good deal of unnecessary monolithic chunkiness in this component. We want page-level components to be super, super, super, super, super lean, basically be a declaration of the top-level pieces and literally nothing else, if we can.

A particular example;

* only the cancel button uses the navigate fn
* yet, we have `useNavigate` in this top-level component
* yet, we have `handleCancel` in this top-level component

Why? This is a distinct concern that we can easily separate out;

```
/**
 * router-centric "cancel" button. We will "cancel"
 * by simply navigating 'back'
 */
function NavCancelButton() {
  const navigate = useNavigate();
  return <button onClick={() => nagivate(-1)}>Cancel</button>
}
```

Now, we have something re-usable, and we pulled some stuff out of this component.

Again, we want page-level components to ONLY declare high-level children. I would love this to look like:

```
function TransactionCreationPage() {
  const [ formData, setFormData ] useState(null);
  const [ shouldShowFeedback , setShouldShowFeedback ] = useState(false);
  return (
    <>
      <TxnPageHeader />
      { shouldShowFeedback && <FeedbackMessage formData={formData} /> }
      <TxnCreateForm
        formData={formData}
        setFormData={setFormData}
        setShouldShowFeedback={setShouldShowFeedback} />
      <NavCancelButton />
    </>
  )
}
```

This way, _so many things_ are immediately made super clear;

* the form can control visibility of feedback, but the feedback component cannot!
* the feedback component has read-only access to the form data; it has no means to update it!
* the nav cancel button is fully independent, and does not depend on page-level state – nice!
* same thing is true for the header